### DISCUSSION TITLE
Envio de arquivo JPG/PNG

### DISCUSSION THREAD

**Discussion by Russimar** [viniciussanchez/RESTRequest4Delphi#93] [Ideas]
Estava com dificuldades de consumir um rota para envio de um arquivo png/jpg, via postman estava funcionando, porém no body tinha que marcar a opção de Binary, vi que no header ele criava o Content-Type do tipo image/png ou image/jpeg, a alternativa que encontrei foi sobreescrever o método addbody fazendo o seguinte:
 function AddBody(const AContent: TStream; const AOwns: Boolean = True; const AContentType: TRESTContentType = ctAPPLICATION_JSON): IRequest; overload;

Sendo assim tive um método para alterar o contentType em tempo de execução e com a opção de enviar um TStream, não sei se é a melhor maneira mas funcionou.
Se alguém já passou por problema semelhante e gostaria de compartilhar, seria interessante.

**Comment by viniciussanchez** [comment]
Legal... você também pode alterar o Content-Type da seguinte forma:

TRequest.New.ContentType('')

Dessa forma não da certo? Dai não precisa alterar o addbody

**Reply by Russimar** [reply]
Boa Vinícius, vou tentar!!!

**Reply by Russimar** [reply]
Assim TRequest.New.ContentType('image/jpeg'), não rolou!!!! Já tinha feito, só não lembrava.
